### PR TITLE
Fix propagation of pod labels on GKE with MIG devices by scanning for GKE device ID format

### PR DESCRIPTION
**Pull request by suffiank**

### CODE REVIEW

**Comment by elfinhe** [comment]
#92

**Comment by suffiank** [comment]
Here's an alternative fix that simply assumes DCGM_EXPORTER_KUBERNETES_GPU_ID_TYPE == "device-name" in case of MIGs will have device IDs match the GKE form "nvidia0/gi7".
https://github.com/NVIDIA/dcgm-exporter/compare/main...suffiank:dcgm-exporter:sufkha/migs-on-gke

I'm not sure how many other device plugins are out there.

File: pkg/dcgmexporter/kubernetes.go [hunk 1]
```diff
@@ -35,6 +36,8 @@ var (
 	socketPath = socketDir + "/kubelet.sock"
 
 	connectionTimeout = 10 * time.Second
+
+	gkeMigDeviceIdRegex = regexp.MustCompile(`^nvidia(?P<gpu>[0-9]+)/gi(?P<gi>[0-9]+)$`)
```
Comment (nikkon-dev) [inline pkg/dcgmexporter/kubernetes.go:40]: You do not use the group names in the code below, but those make the regex visually more difficult to read. Could we remove the group names or use them in the code below?

Comment (suffiank) [reply pkg/dcgmexporter/kubernetes.go]: Yeah, makes sense - removed.

**Review by nikkon-dev — APPROVED** [review]
LGTM

**Pull request merged**